.github/workflows/deploy-baas.yml

```diff
@@ -26,12 +26,20 @@ env:
   DOTNET_NOLOGO: true
 jobs:
   deploy-baas:
-    runs-on: ubuntu-latest
     name: Deploy Cluster
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        differentiator: ${{ fromJson(inputs.differentiators) }}
     outputs:
-      clusterName: ${{ steps.deploy-cluster.outputs.clusterName }}
+      clusterName: ${{ steps.generate-cluster-name.outputs.clusterName }}
     timeout-minutes: 20
     steps:
+    - name: Generate cluster name
+      id: generate-cluster-name
+      run: |
+        name=$( echo ${{ github.run_id }}-${{ github.run_attempt }} | md5sum | cut -c1-7)
+        echo "clusterName=$name" >> "$GITHUB_OUTPUT"
```

Comment:
Why not output the cluster name + the differentiator at this point already? Then we don't need to remember to append the differentiator everywhere else.

Reply:
Oh.. is it because we can't create dynamic job outputs for matrix jobs?

Reply:
Unfortunately yes. At the moment the outputs overwrite each other, so the last one that runs is the one that wins. 
They are working on it though: https://github.com/actions/runner/pull/2477

Reply:
Merged: March, no timeframe for when it'll be actually available, lovely 🥲

Reply:
Yeah, I wouldn't hold my breath 😁